### PR TITLE
Generic autocomplete

### PR DESCRIPTION
**Pull request by IlanSirisky**
Implemented Autocomplete component, Avatar component, and ListItem.
2 interfaces for autocomplete and select options.

### CODE REVIEW

File: src/types/optionsTypes.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface selectOptionsType {
```
Comment (ZivCohenMoveo) [inline src/types/optionsTypes.ts]:
```suggestion
export interface SelectOptionsType {
```

File: src/types/optionsTypes.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  value: string;
+  label: string;
+}
+
+export interface autocompleteOptionsType extends selectOptionsType {
```
Comment (ZivCohenMoveo) [inline src/types/optionsTypes.ts]:
```suggestion
export interface AutocompleteOptionsType extends SelectOptionsType {
```

File: src/components/Autocomplete/Autocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const Autocomplete = ({
+  options,
+  placeholder = "Search",
+  sx,
+}: AutocompleteProps) => {
+  const [inputValue, setInputValue] = useState<string>("");
+
+  return (
+    <MuiAutocomplete
+      sx={{ ...sx }}
```
Comment (ZivCohenMoveo) [inline src/components/Autocomplete/Autocomplete.tsx]:
```suggestion
      sx={sx}
```

File: src/components/Autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      getOptionLabel={(option) => option.label}
+      inputValue={inputValue}
+      onInputChange={(_event, newInputValue) => {
+        setInputValue(newInputValue);
+      }}
+      renderOption={(props, option) => (
+        <AutocompleteListItem
+          option={option}
+          props={props}
+          key={option.value}
```
Comment (ZivCohenMoveo) [inline src/components/Autocomplete/Autocomplete.tsx:34]: not in used

**Review by ZivCohenMoveo — CHANGES_REQUESTED** [review]

File: src/components/Autocomplete/Autocomplete.tsx (hunk 3 above)
Comment (IlanSirisky) [reply src/components/Autocomplete/Autocomplete.tsx]: Applied change

File: src/types/optionsTypes.ts (hunk 1 above)
Comment (IlanSirisky) [reply src/types/optionsTypes.ts]: Applied change

File: src/types/optionsTypes.ts (hunk 2 above)
Comment (IlanSirisky) [reply src/types/optionsTypes.ts]: Applied change

**Pull request merged**